projects/ngx-d3timeline/src/lib/ngx-d3timeline.service.ts

```diff
@@ -79,4 +92,18 @@ export class NgxD3TimelineService implements OnDestroy {
   private zoomed() {
     this.store.dispatch(new fromActions.ZoomedAction(event));
   }
+
+  setupResizing(hostElement: ElementRef, changeDetector: ChangeDetectorRef) {
+    const resizes$ = createResizeObservable(hostElement.nativeElement);
+
+    resizes$
+      .pipe(takeUntil(this.destroySubject), debounceTime(100))
```

Comment:
I think `takeUntil` should be last operator in the pipe.

Reply:
My thinking is if it is like this it will terminate immediately when the service is destroyed. If the debounce is first it will delay the termination unnecessarily.

Reply:
Although there won't be any leaks in this case, its good practice to put the takeuntil operator last to prevent leaks. 

Also I'm pretty sure putting takeUntil last will have the same effect since debounceTime also emits when the source observable completes.

Reply:
Although there won't be any leaks here, its usually best practice to put the takeUntil last to prevent leaks.

Also, in this instance, I think putting the takeUntil last will have the same effect as putting it first. This is because (I think) debounceTime emits a value after the source observable completes.

Reply:
changed, wasn't aware of the ordering, looks like a bit of a potential minefield using this pattern as there are also exceptions where it should be placed first (debounce is not one of them).

https://ncjamieson.com/avoiding-takeuntil-leaks/
Exceptions: https://github.com/cartant/rxjs-tslint-rules/blob/v4.14.2/source/rules/rxjsNoUnsafeTakeuntilRule.ts#L41-L62